eng/pipelines/common/global-build-job.yml

```diff
@@ -19,10 +19,18 @@ jobs:
     pool: ${{ parameters.pool }}
     container: ${{ parameters.container }}
     timeoutInMinutes: ${{ parameters.timeoutInMinutes }}
-    variables: ${{ parameters.variables }}
     workspace:
       clean: all
 
+    variables:
+    - ${{ each variable in parameters.variables }}:
+      - ${{insert}}: ${{ variable }}
```

Comment:
Nit: you don’t need the for each here, you can get this done by inserting the whole object:

```yml
${{ insert }}: ${{ parameters.variables }}
```